mantidproject/mantid
# Add semi axis length options to IntegratePeaksMD for ellipsoid integration

**Description of work.**

This expands the PeakRadius, BackgroundInnerRadius, and BackgroundOuterRadius options in `IntegratePeaksMD` to allow three values to be specified describing the semi-axis lengths of an ellipsoid shape. These can only be specified when the Ellipsoid option is enabled.

**To test:**



Verify that the same result is obtained between normal spherical integration with radius r, and when using Ellipsoid=True with a PeakRadius of r, r, r.

Run the unit tests  `MDAlgorithmsTest IntegratePeaksMD2Test` to make sure they pass.

*There is no associated issue.*

The version into `ornl-next` - #30431

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
